python/ql/src/Functions/ModificationOfParameterWithDefault.ql

```diff
@@ -93,7 +93,47 @@ class Mutation extends TaintSink {
     }
 }
 
-from TaintedPathSource src, TaintedPathSink sink
-where src.flowsTo(sink)
+/** copying prevents modification of the default value */
+class Copying extends Sanitizer {
+    Copying() { this = "Copy sanitizer"}
+
+    override predicate sanitizingDefinition(TaintKind kind, EssaDefinition def) {
+        (kind instanceof EmptyMutableValue
+        or
+        kind instanceof NonEmptyMutableValue) and
+        isCopied(def.(AssignmentDefinition).getValue())
+    }
+
+    predicate isCopied(ControlFlowNode value) {
+        value = Value::named("copy.copy").getACall()
+        or
+        value = Value::named("copy.deepcopy").getACall()
+        or
+        value.(AttrNode).getName() = "copy"
+        or
+        value.(CallNode).getNode().getFunc().(Attribute).getName() = "copy"
```

Comment:
Going through the AST is unnecessary here. `CallNode`s have a `getFunction` method.